Unify native file name props

am11 opened the pull request:
The top level `Directory.Build.props` imports eng/native/naming.props,
which defines platform specific naming conventions. Use that in places
where locally defined props are used and delete the redundant props.

lambdageek commented on src/mono/mono.proj:
```diff
@@ -14,29 +14,21 @@
     <MonoSharedLibName Condition="'$(MonoSharedLibName)' == ''">$(CoreClrLibName)</MonoSharedLibName>
-    <MonoSharedLibFileName>$(LibPrefix)$(MonoSharedLibName)$(SharedLibExt)</MonoSharedLibFileName>
-    <MonoStaticLibFileName>$(LibPrefix)$(MonoLibName)$(StaticLibExt)</MonoStaticLibFileName>
+    <MonoSharedLibFileName>$(LibPrefix)$(MonoSharedLibName)$(LibSuffix)</MonoSharedLibFileName>
+    <MonoStaticLibFileName>$(LibPrefix)$(MonoLibName)$(LibSuffix)</MonoStaticLibFileName>
```
This doesn't seem right: we care about shared/static lib differences.

am11 replied:
Yup, it's a typo I'm fixing it. (we have `StaticLibSuffix`)

lambdageek commented on src/mono/mono.proj:
```diff
@@ -14,29 +14,21 @@
     <ScriptExt Condition="'$(OS)' == 'Windows_NT'">.cmd</ScriptExt>
     <ScriptExt Condition="'$(OS)' != 'Windows_NT'">.sh</ScriptExt>
-    <ExeExt Condition="'$(OS)' == 'Windows_NT'">.exe</ExeExt>
-    <LibPrefix Condition="'$(TargetsWindows)' != 'true'">lib</LibPrefix>
-    <SharedLibExt Condition="'$(TargetsOSX)' == 'true' or '$(TargetsMacCatalyst)' == 'true' or '$(TargetsiOS)' == 'true' or '$(TargetstvOS)' == 'true'">.dylib</SharedLibExt>
-    <SharedLibExt Condition="'$(TargetsWindows)' == 'true'">.dll</SharedLibExt>
-    <SharedLibExt Condition="'$(SharedLibExt)' == ''">.so</SharedLibExt>
-    <StaticLibExt Condition="'$(TargetsWindows)' == 'true'">.lib</StaticLibExt>
-    <StaticLibExt Condition="'$(StaticLibExt)' == ''">.a</StaticLibExt>
```
Is `LibSuffix` going to be the right thing for our cross-compilation targets? (ie will this play nice with src/mono/monoaotcross.proj ?) /cc @directhex

am11 replied:
The conditions in naming/props are based on `PackageRID`, which undergoes complicated calculation which accounts for cross, portable and other aspects of matrix: https://github.com/dotnet/runtime/blob/ca072ba73c92e5730b382806291e6d1b48d7fbfd/Directory.Build.props#L158
Lets fix it here if something is missing (we will find out by the CI run).

steveisok replied:
@directhex Please make sure to validate the official build to make sure we don't have any cross-compilation packaging issues.

akoeplinger replied:
I've scheduled an official build here: https://dev.azure.com/dnceng/internal/_build/results?buildId=1512127&view=results

akoeplinger replied:
The official build is green so we should be good here :)

am11 commented on src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets:
```diff
@@ -80,8 +73,8 @@ The .NET Foundation licenses this file to you under the MIT license.
-    <SharedLibrary Condition="'$(TargetOS)' == 'windows'">$(FrameworkLibPath)\Framework$(LibFileExt)</SharedLibrary>
-    <SharedLibrary Condition="'$(TargetOS)' != 'windows'">$(FrameworkLibPath)\libframework$(LibFileExt)</SharedLibrary>
+    <SharedLibrary Condition="'$(TargetOS)' == 'windows'">$(FrameworkLibPath)\Framework$(StaticLibSuffix)</SharedLibrary>
+    <SharedLibrary Condition="'$(TargetOS)' != 'windows'">$(FrameworkLibPath)\libframework$(StaticLibSuffix)</SharedLibrary>
```
@jkotas, the name of this property has `shared` but the actual value of file extension was (and is) was static: ('.lib' for windows and '.a' for unix). Is this a typo in the property name?

jkotas replied:
I do not think it is a typo. "shared" here means "shared between multiple projects".

jkotas commented on src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets:
```diff
@@ -51,20 +51,13 @@ The .NET Foundation licenses this file to you under the MIT license.
     <NativeObjectExt Condition="'$(TargetOS)' == 'windows'">.obj</NativeObjectExt>
     <NativeObjectExt Condition="'$(TargetOS)' != 'windows'">.o</NativeObjectExt>
 
-    <LibFileExt Condition="'$(TargetOS)' == 'windows'">.lib</LibFileExt>
```
This file is copied into the NativeAOT package. @MichalStrehovsky have not brought the packaging infrastructure over yet, so it is not easy to see.

If it were to depend on other files in the repo, those other files need to be copied into the NativeAOT package as well that gets complex fast. I think it is better to leave it self-contained.

am11 replied:
Makes sense. I've reverted NativeAOT changes.

am11 committed:
Unify native file name props
The top level `Directory.Build.props` imports eng/native/naming.props,
which defines platform specific naming conventions. Use that in places
where locally defined props are used and delete the redundant props.

akoeplinger commented on Directory.Build.props:
```diff
@@ -17,6 +17,7 @@
     <_hostOS Condition="$([MSBuild]::IsOSPlatform('SOLARIS'))">Solaris</_hostOS>
     <_hostOS Condition="$([MSBuild]::IsOSPlatform('WINDOWS'))">windows</_hostOS>
+    <HostOS>$(_hostOS)</HostOS>
     <_hostOS Condition="'$(TargetOS)' == 'Browser'">Browser</_hostOS>
```
this can actually be deleted, there's no way we'll be building on a "Browser" host OS I think.

am11 replied:
we can't just delete it because in case of "browser" the definitions of what is host and what is target does not fit the bill. other build systems and toolchains have a third dimension for it called "build", which we don't support. so here we misuse this (internal) property name in this case.

you can try to delete this line on main to see how it breaks the wasm packaging

akoeplinger commented on Directory.Build.props:
```diff
@@ -17,6 +17,7 @@
     <_hostOS Condition="$([MSBuild]::IsOSPlatform('ILLUMOS'))">illumos</_hostOS>
     <_hostOS Condition="$([MSBuild]::IsOSPlatform('SOLARIS'))">Solaris</_hostOS>
     <_hostOS Condition="$([MSBuild]::IsOSPlatform('WINDOWS'))">windows</_hostOS>
+    <HostOS>$(_hostOS)</HostOS>
```
I think it'd be cleaner to set `HostOS` directly rather than going through the `_hostOS` indirection.

am11 replied:
Yup, it would be cleaner but we would still need a new property for that kind of a cleanup, which can be done separately.

akoeplinger commented on Directory.Build.props:
```diff
@@ -162,13 +163,13 @@
-    <_toolRuntimeRID Condition="'$(_runtimeOS)' == 'browser' and $([MSBuild]::IsOSPlatform('WINDOWS'))">win-x64</_toolRuntimeRID>
-    <_toolRuntimeRID Condition="'$(_runtimeOS)' == 'browser' and $([MSBuild]::IsOSPlatform('OSX'))">osx-x64</_toolRuntimeRID>
+    <_toolRuntimeRID Condition="'$(_runtimeOS)' == 'browser' and '$(_hostOS)' == 'windows'">win-x64</_toolRuntimeRID>
+    <_toolRuntimeRID Condition="'$(_runtimeOS)' == 'browser' and '$(_hostOS)' == 'osx'">osx-x64</_toolRuntimeRID>
```
use `HostOS` here

am11 replied:
yea, i was waiting for builds to finish for the next round of changes. it won't be ready until the CI is green.

akoeplinger commented on eng/native/naming.props:
```diff
@@ -5,25 +5,25 @@
       </PropertyGroup>
     </When>
-    <When Condition="$(PackageRID.StartsWith('osx'))">
+    <When Condition="$(PackageRID.StartsWith('osx')) or $(PackageRID.StartsWith('mac')) or $(PackageRID.StartsWith('ios')) or $(PackageRID.StartsWith('tvos'))">
```
to make it more explicit:

```suggestion
    <When Condition="$(PackageRID.StartsWith('osx')) or $(PackageRID.StartsWith('maccatalyst')) or $(PackageRID.StartsWith('ios')) or $(PackageRID.StartsWith('tvos'))">
```

akoeplinger commented on src/mono/mono.proj:
```diff
@@ -14,29 +14,21 @@
     <MonoForceInterpreter Condition="'$(MonoForceInterpreter)' == ''">false</MonoForceInterpreter>
-    <ExeExt Condition="'$(OS)' == 'Windows_NT'">.exe</ExeExt>
     <ScriptExt Condition="'$(OS)' == 'Windows_NT'">.cmd</ScriptExt>
     <ScriptExt Condition="'$(OS)' != 'Windows_NT'">.sh</ScriptExt>
```
could use `HostOS` here too

am11 committed:
Use HostOS to determine windows browser-wasm case
